app/controllers/journalist/articles_controller.rb

```diff
@@ -3,6 +3,11 @@ class Journalist::ArticlesController < ApplicationController
     before_action :authenticate_user!
     before_action :check_author, only: [:edit, :destroy, :update]
 
+    def index
+        @articles = Article.all.where(user_id: current_user.id)
+        @notapproved = Article.all.where(user_id: current_user.id, approved: false) 
```

Comment:
`@notapproved = current_user.articles.where(approved: false)`
with this you dont have loop through all articles in the database.

Reply:
done